crates/bitwarden-core/src/key_management/crypto.rs

```diff
@@ -574,76 +584,88 @@ pub(super) fn verify_asymmetric_keys(
     })
 }
 
-/// A new signing key pair along with the signed public key
+/// Response for the `make_keys_for_user_crypto_v2`, containing a set of keys for a user
 #[derive(Serialize, Deserialize, Debug, JsonSchema)]
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
 #[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
 #[cfg_attr(feature = "wasm", derive(Tsify), tsify(into_wasm_abi, from_wasm_abi))]
-pub struct MakeUserSigningKeysResponse {
-    /// Base64 encoded verifying key
-    verifying_key: String,
-    /// Signing key, encrypted with the user's symmetric key
-    signing_key: EncString,
+pub struct UserCryptoV2KeysResponse {
+    /// User key
+    user_key: String,
+
+    /// Wrapped private key
+    private_key: EncString,
+    /// Public key
+    public_key: String,
     /// The user's public key, signed by the signing key
     signed_public_key: SignedPublicKey,
+
+    /// Signing key, encrypted with the user's symmetric key
+    signing_key: EncString,
+    /// Base64 encoded verifying key
+    verifying_key: String,
+
+    /// The user's signed security state
+    security_state: SignedSecurityState,
+    /// The security state's version
+    security_version: u64,
 }
 
-/// Makes a new set of signing keys for a user, which should only be done during
-/// once. This also signs the public key with the signing key
-/// and returns the signed public key.
-pub fn make_user_signing_keys_for_enrollment(
+/// Initializes the user's cryptographic state for v2 users.
+/// If the client already contains a v1 user, then this user's private-key will be
+/// re-used.
+pub(crate) fn make_keys_for_user_crypto_v2(
     client: &Client,
-) -> Result<MakeUserSigningKeysResponse, CryptoError> {
+) -> Result<UserCryptoV2KeysResponse, CryptoError> {
     let key_store = client.internal.get_key_store();
     let mut ctx = key_store.context();
 
-    // Make new keypair and sign the public key with it
-    let signature_keypair = SigningKey::make(SignatureAlgorithm::Ed25519);
-    let temporary_signature_keypair_id = SigningKeyId::Local("temporary_key_for_rotation");
+    let temporary_user_key_id = SymmetricKeyId::Local("temporary_user_key");
+    let temporary_signing_key_id = SigningKeyId::Local("temporary_signing_key");
+    let temporary_private_key_id = AsymmetricKeyId::Local("temporary_private_key");
```

Comment:
This could be simplified a bit with #274, we might want to try to prioritize it.